### PR TITLE
chore: [CDS-69888]: add jest test cases for AwsSamBuild and AwsSamDeploy steps

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
### Summary
- Test cases cover use cases of both, edit and runtime views.

#### Screenshots
**AwsSamBuild:**
Screenshot 2023-07-21 at 7 57 03 PM

**AwsSamDeploy:**
Screenshot 2023-07-21 at 7 55 52 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineSteps/Steps/StepGroupStep/__tests__/StepGroupStepEdit.test.tsx [hunk 1]
```diff
@@ -35,14 +33,6 @@ jest.mock('services/portal', () => ({
 }))
 
 describe('StepGroupStepEdit tests', () => {
-  const stepGroupStep = new StepGroupStep()
-  beforeAll(() => {
-    factory.registerStep(stepGroupStep)
-  })
-  afterAll(() => {
-    factory.deregisterStep(stepGroupStep.getType())
-  })
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/StepGroupStep/__tests__/StepGroupStepEdit.test.tsx:44]: Why is this removed?

**Review by rameshx — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/StepGroupStep/__tests__/StepGroupStepEdit.test.tsx (hunk 1 above)
Comment (chetanvekariya-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/StepGroupStep/__tests__/StepGroupStepEdit.test.tsx:44]: Not needed. We are using edit component directly to test it in this test file.
